model/lifecycle.go

```diff
@@ -1442,14 +1442,26 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskname := range pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
+				if !t.Activated && activated {
+					// update task activation
+					if _, err = t.ActivateTask(evergreen.User); err != nil {
+						return nil, errors.Wrapf(err, "problem updating active state for existing task '%s'", taskname)
+					}
```

Comment:
In the first commit I included an update to the task in the map here, but actually task names doesn't allow for duplicates so we don't have to worry about it.